Source/EasyNetQ/ConnectionConfigurationExtensions.cs

```diff
@@ -41,34 +41,51 @@ public static void SetDefaultProperties(this ConnectionConfiguration configurati
             {
             }
 
-        var hostname = Environment.MachineName;
-
-        var netVersion = Environment.Version.ToString();
-        configuration.Product ??= applicationName;
-        configuration.Platform ??= hostname;
-        configuration.Name ??= applicationName;
-
         AddValueIfNotExists(configuration.ClientProperties, "client_api", "EasyNetQ");
-        AddValueIfNotExists(configuration.ClientProperties, "product", configuration.Product);
-        AddValueIfNotExists(configuration.ClientProperties, "platform", configuration.Platform);
-        AddValueIfNotExists(configuration.ClientProperties, "net_version", netVersion);
-        AddValueIfNotExists(configuration.ClientProperties, "version", version);
-        AddValueIfNotExists(configuration.ClientProperties, "easynetq_version", version);
+        AddValueIfNotExists(configuration.ClientProperties, "product", configuration.Product ?? applicationName);
+        AddValueIfNotExists(configuration.ClientProperties, "platform", configuration.Platform ?? GetPlatform());
+        AddValueIfNotExists(configuration.ClientProperties, "os", Environment.OSVersion.ToString());
+        AddValueIfNotExists(configuration.ClientProperties, "version", GetApplicationVersion());
+        AddValueIfNotExists(configuration.ClientProperties, "connection_name", configuration.Name ?? applicationName);
+        AddValueIfNotExists(configuration.ClientProperties, "easynetq_version", typeof(ConnectionConfigurationExtensions).Assembly.GetName().Version.ToString());
         AddValueIfNotExists(configuration.ClientProperties, "application", applicationName);
         AddValueIfNotExists(configuration.ClientProperties, "application_location", applicationPath);
-        AddValueIfNotExists(configuration.ClientProperties, "machine_name", hostname);
+        AddValueIfNotExists(configuration.ClientProperties, "machine_name", Environment.MachineName);
+        AddValueIfNotExists(configuration.ClientProperties, "user", configuration.UserName);
```

Comment:
I added back `user`, `connected` and `requested_heartbeat` properties. I do not know why they were removed since my last commit. Let me know if those properties were deleted deliberately.